### DISCUSSION TITLE
Nuxt authenticate Middleware doesn't work when click back button from browser

### DISCUSSION THREAD

**Discussion by umairali02** [nuxt/nuxt#10683] [Questions (Nuxt 2)]
I'm implementing the protected route in nuxt the best way I found is using the middleware, so I implemented it with the using of module **cookie-universal-nuxt.**

Everything is working smoothly, but when user session is expire and I click on any protected route its navigating me to the login page which is what I want the only bug is that when I click on the back button on the browser (browser default back button) it shows again the protected page while the user session is already expire.

**setting the cookies and in localstroage and cookies.**

```
const expireAt = Helpers.getJwtData().exp * 1000
window.localStorage.setItem('tokenExpire',expireAt);
this.$cookies.set('tokenExpire', expireAt, { path: '/'})
```

middleware/autheticate.js
```
export default function (context) {
  const expiry = context.$cookies?.get('tokenExpire');
  const isLoggedIn = +expiry > new Date().getTime
  if (!isLoggedIn) {
    Helpers.logout();
    context.redirect(`/login?redirect_path=${context.route.path}`);
  }
}
```

**Comment by gouz7514** [comment]
do you have any solution? I'm facing same issue :(

**Comment by oiluis** [comment]
Bump ... same issue ... wondering if you guys were able to solve it after few years later ...
